libraries/shared/storage/extractor_test.go

```diff
@@ -0,0 +1,65 @@
+package storage_test
+
+import (
+	"math/rand"
+
+	"github.com/makerdao/vulcanizedb/libraries/shared/mocks"
+	"github.com/makerdao/vulcanizedb/libraries/shared/storage"
+	"github.com/makerdao/vulcanizedb/libraries/shared/storage/types"
+	"github.com/makerdao/vulcanizedb/libraries/shared/test_data"
+	"github.com/makerdao/vulcanizedb/pkg/fakes"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Storage diff extractor", func() {
+	var (
+		mockFetcher    *mocks.MockStorageFetcher
+		mockRepository *mocks.MockStorageDiffRepository
+		extractor      storage.DiffExtractor
+	)
+
+	BeforeEach(func() {
+		mockFetcher = mocks.NewMockStorageFetcher()
+		mockRepository = &mocks.MockStorageDiffRepository{}
+		extractor = storage.DiffExtractor{
+			StorageDiffRepository: mockRepository,
+			StorageFetcher:        mockFetcher,
+		}
+	})
+
+	Describe("ExtractDiffs", func() {
+		It("fetches storage diffs", func() {
+			mockFetcher.ErrsToReturn = []error{fakes.FakeError}
+
+			_ = extractor.ExtractDiffs()
+
+			Expect(mockFetcher.FetchStorageDiffsCalled).To(BeTrue())
+		})
+
+		It("returns error if fetching storage diffs fails", func() {
+			mockFetcher.ErrsToReturn = []error{fakes.FakeError}
+
+			err := extractor.ExtractDiffs()
+
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(MatchError(fakes.FakeError))
```

Comment:
We probably don't need both, but I'm a fan because of the clearer failure message (`expected err to have occurred, got none` vs `expected nil to equal "Failed"`). Not super wedded to this, but I think I'd opt for a more comprehensive approach if we decide to minimize our assertions throughout the test suites

Reply:
Nice, yeah I agree that isn't a very good test failure if the error didn't occur. Especially if we're using this pattern elsewhere I def have no problem leaving it like this